mlflow/utils/search_utils.py

```diff
@@ -322,3 +324,45 @@ def sort(cls, runs, order_by_list):
                           key=lambda run: cls._get_value_for_sort(run, key_type, key, ascending),
                           reverse=not ascending)
         return runs
+
+    @classmethod
+    def _parse_offset_from_page_token(cls, page_token):
+        # Note: the page_token is expected to be a base64-encoded JSON that looks like
+        # { "offset": xxx }. However, this format is not stable, so it should not be
+        # relied upon outside of this method.
+        if not page_token:
+            return 0
+
+        try:
+            decoded_token = base64.b64decode(page_token)
+        except TypeError:
+            raise MlflowException("Invalid page token, could not base64-decode",
+                                  error_code=INVALID_PARAMETER_VALUE)
+
+        try:
+            parsed_token = json.loads(decoded_token)
+        except TypeError:
+            raise MlflowException("Invalid page token, decoded value=%s" % decoded_token,
+                                  error_code=INVALID_PARAMETER_VALUE)
+
+        offset_str = parsed_token.get("offset")
+        if not offset_str:
+            raise MlflowException("Invalid page token, parsed value=%s" % parsed_token,
```

Comment:
Subtly different error messages allow us to pinpoint what failed, without revealing information about the expected internal structure of the token (which is not an API) :)